sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -193,6 +196,9 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
     case OneRowRelation =>
       ""
 
+    case Hint(_, _, child) =>
```

Comment:
Also, `SparkStrategies` ensures that here https://github.com/apache/spark/pull/14132/files#diff-7253a38df7e111ecf6b1ef71feba383bR347 , too.